presto-main/src/main/java/com/facebook/presto/memory/NodeMemoryConfig.java

```diff
@@ -29,6 +31,7 @@
 
     // This is a per-query limit for the user plus system allocations.
     private DataSize maxQueryTotalMemoryPerNode = new DataSize(Runtime.getRuntime().maxMemory(), BYTE);
+    private double heapHeadroom;
```

Comment:
Find out what an appropriate initial values. Also check with @martint and see what his opinion is on percentage vs absolute value.